### PR TITLE
Support Decimal type MongoDB

### PR DESCRIPTION
**Pull request by Amedeo91**
This PR will provide support for mongodb decimal data type: #1045.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,10 +12,19 @@ python:
   - pypy
 install: travis_retry pip install tox-travis
 services:
-  - mongodb
+  #- mongodb
   - redis-server
 before_script:
+  # work-around to make travis-ci working with mongod 3.4
```
Comment (nicolaiarocci) [inline .travis.yml]: please keep an eye on travis-ci and submit a PR when the work-around will not be needed anymore

File: CHANGES [hunk 2]
```diff
@@ -8,6 +8,8 @@ Development
 
 Version 0.8
 ~~~~~~~~~~~
+- Updating to pymongo 3.5 and supporting new mongodb decimal data-type
```
Comment (nicolaiarocci) [inline CHANGES]: please split that in two items: 

```
Upgrade: PyMongo 3.5
New:  add support for new MongoDB decimal data type etc.
```

File: docs/validation.rst [hunk 3]
```diff
@@ -47,7 +47,11 @@ Extending Data Validation
 Data validation is based on the Cerberus_ validation system and it is therefore
 extensible. As a matter of fact, Eve's MongoDB data-layer itself extends
 Cerberus validation, implementing the ``unique`` and ``data_relation``
-constraints and the ``ObjectId`` data type on top of the standard rules.
+constraints, the ``ObjectId`` data type and the ``decimal128`` on top of
+the standard rules.
+
+.. note::
```
Comment (nicolaiarocci) [inline docs/validation.rst]: this note should not be needed, as we're adding `decimal` in the supported types (see config.rst changes)

File: eve/io/mongo/mongo.py [hunk 4]
```diff
@@ -53,6 +55,9 @@ def default(self, obj):
             if obj.database:
                 retval['$db'] = obj.database
             return retval
+        if isinstance(obj, decimal128.Decimal128):
+            # If I store a big decimal, I need to parse it as string
```
Comment (nicolaiarocci) [inline eve/io/mongo/mongo.py]: comment not needed

File: .travis.yml (hunk 1 above)
Comment (Amedeo91) [reply .travis.yml]: Yes I will, but I do not think it will happen any soon!

File: CHANGES (hunk 2 above)
Comment (Amedeo91) [reply CHANGES]: Ok

**Pull request merged**